lib/encoder-stream.js

```diff
@@ -19,6 +21,9 @@ class EncoderStream extends Transform {
 	}
 
 	_transform(object, encoding, callback) {
+		const timestamp = Number.parseFloat(object.tags.AWS_KINESISVIDEO_PRODUCER_TIMESTAMP) + (this.count * (this.fps / 1000));
```

Comment:
this is strange for me.

First i do not understand why we are doing this in the encoder, i feel it should be done in the decoder, because this is where you have the `keyframeOnly`, and the 'fps' option is an alternative to `keyframeOnly`

I feel that object.tags.AWS_KINESISVIDEO_PRODUCER_TIMESTAMP is correct when the frame is a keyframe.

if you do like this, it means that you will override the AWS_KINESISVIDEO_PRODUCER_TIMESTAMP when it is correct.

I suggest you store 2 objects `this.lastKeyframeTimestamp` and `this.countSinceLastKeyframe` object, and do the following : 
* first build `incrementalTimestamp` from `this.lastKeyframeTimestamp`, `this.countSinceLastKeyframe` and `this.fps`
* Then 
  * when the frame is not a keyframe, 
    * check that AWS_KINESISVIDEO_PRODUCER_TIMESTAMP is same than  lastKeyframeTimestamp (if not raise an error) 
    * override AWS_KINESISVIDEO_PRODUCER_TIMESTAMP with `incrementalTimestamp`
  * when the frame is a keyframe, 
    * check that AWS_KINESISVIDEO_PRODUCER_TIMESTAMP is different than lastKeyframeTimestamp (if not raise an error)
    * check that AWS_KINESISVIDEO_PRODUCER_TIMESTAMP is strictly greater than the last incrementalTimestamp (if not raise an error)
    * set `this.countSinceLastKeyframe` to 0
    * set `this.lastKeyframeTimestamp` to AWS_KINESISVIDEO_PRODUCER_TIMESTAMP